react/features/base/media/middleware.js

```diff
@@ -25,6 +30,9 @@ const logger = require('jitsi-meet-logger').getLogger(__filename);
  */
 MiddlewareRegistry.register(store => next => action => {
     switch (action.type) {
+    case CONFERENCE_LEFT:
+        _clearVideoTransforms(store);
+        break;
```

Comment:
1. Please separate `case` blocks with an empty line to visually aid the understanding of the fact that they are unrelated. I think I saw them bunched up in multiple files in this PR.

2. I presume it's for the sake of the abstraction's completeness but given all the other complexity around transforms I could've lived without the action `CLEAR_VIDEO_TRANSFORMS` I guess. The reducer could've directly reduced the actions which currently cause `CLEAR_VIDEO_TRANSFORMS` to be dispatched.

3. Why only `CONFERENCE_LEFT`? Why not `CONFERENCE_FAILED` as well?

4. Why clear all "video transform"s at the end of the conference? Why not clear when the stream associated with the "video transform" is no longer in the redux state? Or when the associated participant leaves?

Reply:
1. Ok, thanks!
2. Also, ok, I'll move them into the reducer, thanks!
3. Because I didn't know all the events and what they do and when. It would be good to have an extracted documentation about them. Maybe I'll spend some time having one.
4. Pawel also suggested this and I told him I take a look at that action, but as far as I checked (twice) there is no stream ID in that action. Am I missing something?

Reply:
>Pawel also suggested this and I told him I take a look at that action, but as far as I checked (twice) there is no stream ID in that action. Am I missing something?

There must be access to stream ID from TRACK_REMOVED event, have you checked good enough ? (I'll take a look now)

Reply:
There is a JitsiTrack in TRACK_REMOVED even which should be enough to get stream id...

Reply:
Found the stream.id in the jitsiTrack object, thanks! Btw it wasn't easy, as that's a huge object. Is there an easy way to find out what that object is consist of? E.g. some nice comments or an easy to read constructor?